apps/web/styles/globals.css

```diff
@@ -42,6 +42,14 @@ button[role="switch"][data-state="checked"] span {
   transform: translateX(16px);
 }
 
+/* Radix Dropdown has crazy high z index which can't be controlled yet [ref: https://github.com/radix-ui/primitives/issues/1303] 
+   without directly targetting data-radix-portal using CSS.
+   Until they do, we can use the following to make sure dropdown doesn't come above everything else such as a dialog, for instance.
+*/
+[data-radix-portal] {
```

Comment:
Theoretically, it might break it where dropdown comes in portal mode in a dialog. 
Shouldn't we close the dropdown when the item in it has been clicked ? That should solve the problem of Dialog coming below Dropdown.

Reply:
🤔 
Good point. I think in terms of flow, it makes sense to keep the dropdown open when a dialog pops up _after_ the dropdown button click because it suggests that the action is still not complete, in case they wish to cancel it. But I think @Jaibles would be the best man to suggest what's the right course of action in such a case.

Reply:
But the weird part is, we still close it as soon as one of the action buttons in the dialog is clicked, which kind of defeats the purpose of keeping it open? @hariombalhara

Reply:
Once you click **Delete** I would expect that Dropdown to close.

Reply:
After spending a couple of hours on this, there's a cyclic issue. We need to use event's `preventDefault` to ensure that the dialog shows up and remains showing. But this also means that the default behaviour of the dropdown collapsing will be prevented as well. If we don't use this, then dialog doesn't remain on top. 🤔 
Any suggestions @hariombalhara @emrysal ?

Reply:
Also, while we're at it, there's a radix UI component specifically for confirmation dialogs called `AlertDialog`. What are your thoughts on it? @emrysal [[here](https://www.radix-ui.com/docs/primitives/components/alert-dialog)]